backend/app/services/workspace_archive/archive_service.py

```
@@ -0,0 +1,376 @@
+            archive_info = task_crd.status.archive if task_crd.status else None
+
+            if not archive_info or not archive_info.storageKey:
+                return False, None
+
+            # Check expiration
+            if archive_info.expiresAt and archive_info.expiresAt < datetime.utcnow():
+                return False, None
+
+            return True, archive_info.storageKey
+
+        except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)